File: src/main/java/com/freya02/botcommands/api/pagination/interactive/InteractiveMenuBuilder.java
package com.freya02.botcommands.api.pagination.interactive;

import com.freya02.botcommands.api.pagination.PaginatorSupplier;
import org.jetbrains.annotations.NotNull;

/**
 * Builds an {@link InteractiveMenu}
 */
public final class InteractiveMenuBuilder extends BasicInteractiveMenuBuilder<InteractiveMenuBuilder, InteractiveMenu> {
	@Override
	@NotNull
	public InteractiveMenu build() {
		return new InteractiveMenu(constraints, timeout, hasDeleteButton, firstContent, previousContent, nextContent, lastContent, deleteContent, items, usePaginator);
	}

	@Override
	public InteractiveMenuBuilder setPaginatorSupplier(@NotNull PaginatorSupplier<InteractiveMenu> paginatorSupplier) {
		throw new IllegalStateException("Interactive menu builder cannot have a PaginatorSupplier");
	}
}

File: src/main/java/io/github/freya022/botcommands/api/pagination/interactive/InteractiveMenuBuilder.kt
package io.github.freya022.botcommands.api.pagination.interactive

import io.github.freya022.botcommands.api.components.Components
import io.github.freya022.botcommands.api.pagination.PaginatorSupplier

/**
 * Builds an [InteractiveMenu]
 */
class InteractiveMenuBuilder internal constructor(
    componentsService: Components
) : BasicInteractiveMenuBuilder<InteractiveMenuBuilder, InteractiveMenu>(componentsService) {
    override fun build(): InteractiveMenu = InteractiveMenu(componentsService, this)

    override fun setPaginatorSupplier(paginatorSupplier: PaginatorSupplier<InteractiveMenu>): Nothing {
        throw IllegalStateException("Interactive menu builder cannot have a PaginatorSupplier")
    }
}
